src/AutoRest.CSharp/Mgmt/AutoRest/MgmtContext.cs

```diff
@@ -16,6 +17,8 @@ internal static class MgmtContext
 
         public static MgmtOutputLibrary Library => Context.Library;
 
+        public static TypeFactory TypeFactory => Context.TypeFactory;
```

Comment:
I am OK with this change, but the DpgLibrary just has a public property `TypeFactory` for the type factory it is using.
Mgmt could follow the same pattern and do not have to change this.
But that might be a big irrelavant change, it is fine here.

Reply:
Since we already have created an instance of TypeFactory in [context](https://github.com/Azure/autorest.csharp/blob/63b56fd85c988fe795e1ed7829fbad1f09a35c18/src/AutoRest.CSharp/Common/Output/Models/Types/BuildContextOfT.cs#L46), why create another instance [here](https://github.com/Azure/autorest.csharp/blob/63b56fd85c988fe795e1ed7829fbad1f09a35c18/src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs#L33) again?

Reply:
Dpg and HLC is not using BuildContext.
Mgmt is the only target that is using the BuildContext.
We have a plan to unify them, therefore in the near future, Mgmt will no longer use the BuildContext as well.

Reply:
OK, thanks for the info. Sounds like we can leave the refactoring till then.